provision/infinispan/create_ispn_clusters.sh

```diff
@@ -0,0 +1,315 @@
+
+  # Creates caches on site A
+  deploy_all_caches "${KUBECONFIG_1}" "${NS_1}" "${site2}"
+
+  # Create caches on site B
+  deploy_all_caches "${KUBECONFIG_1}" "${NS_2}" "${site1}"
+}
+
+function create_cross_site_multiple_clusters() {
+  # Login on both clusters
+  rosa_oc_login "${KUBECONFIG_1}" "${CLUSTER_1}"
+  rosa_oc_login "${KUBECONFIG_2}" "${CLUSTER_2}"
+
+  # Check and create the namepsaces if missing
+  KUBECONFIG="${KUBECONFIG_1}" oc new-project "${NS_1}" || true
+  KUBECONFIG="${KUBECONFIG_2}" oc new-project "${NS_2}" || true
+
+  # Create secrets for TLS (Openshift Route)
+  create_tls_secrets "${KUBECONFIG_1}" "${NS_1}"
+  create_tls_secrets "${KUBECONFIG_2}" "${NS_2}"
+
+  # Create and share access tokens
+  create_service_account "${KUBECONFIG_1}" "${NS_1}"
+  create_service_account "${KUBECONFIG_2}" "${NS_2}"
+
+  local token1="$(get_service_account_token "${KUBECONFIG_1}" "${NS_1}")"
+  local token2="$(get_service_account_token "${KUBECONFIG_2}" "${NS_2}")"
+
+  create_token_secret "${KUBECONFIG_1}" "${NS_1}" "${XSITE_TOKEN_SECRET}" "${token2}"
+  create_token_secret "${KUBECONFIG_2}" "${NS_2}" "${XSITE_TOKEN_SECRET}" "${token1}"
+
+  local api_url_1="openshift://$(get_api_url "${KUBECONFIG_1}")"
+  local api_url_2="openshift://$(get_api_url "${KUBECONFIG_2}")"
+
+  # Use cluster name as site name
+  local site1="${CLUSTER_1}"
+  local site2="${CLUSTER_2}"
+
+  deploy_infinispan_cr "${KUBECONFIG_1}" "${NS_1}" "${site1}" "${NS_2}" "${site2}" "${api_url_2}"
+  deploy_infinispan_cr "${KUBECONFIG_2}" "${NS_2}" "${site2}" "${NS_1}" "${site1}" "${api_url_1}"
+
+   # Creates caches on site A
+   deploy_all_caches "${KUBECONFIG_1}" "${NS_1}" "${site2}"
+
+   # Create caches on site B
+   deploy_all_caches "${KUBECONFIG_1}" "${NS_2}" "${site1}"
+}
+
+function create_cluster_without_cross_site() {
+  # TLS not required for single cluster cross-site
+  # Login in cluster
+  rosa_oc_login "$KUBECONFIG_1" "${CLUSTER_1}"
+
+  # Check and create the namepsaces if missing
+  KUBECONFIG="${KUBECONFIG_1}" oc new-project "${NS_1}" || true
+
+  # Deploy an Infinispan cluster in each of the namespaces.
+  deploy_infinispan_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}"
+
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "sessions"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "authenticationSessions"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "actionTokens"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "offlineSessions"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "clientSessions"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "offlineClientSessions"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "loginFailures"
+  deploy_cache_cr_without_cross_site "${KUBECONFIG_1}" "${NS_1}" "work"
+}
+
+########
+# MAIN #
+########
+
+if [ "${CLUSTER_1}" == "${CLUSTER_2}" ]; then
```

Comment:
To save having to set both variables, should we just check if CLUSTER_2 is unset/empty?

Reply:
:+1: